src/main/java/org/opensearch/security/hasher/PasswordHasherFactory.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.hasher;
+
+import org.opensearch.common.settings.Settings;
+import org.opensearch.security.support.ConfigConstants;
+
+import static org.opensearch.security.support.ConfigConstants.BCRYPT;
+import static org.opensearch.security.support.ConfigConstants.PBKDF2;
+
+public class PasswordHasherFactory {
+
+    public static PasswordHasher createPasswordHasher(Settings settings) {
+
+        String algorithm = settings.get(
+            ConfigConstants.SECURITY_PASSWORD_HASHING_ALGORITHM,
+            ConfigConstants.SECURITY_PASSWORD_HASHING_ALGORITHM_DEFAULT
+        );
+
+        PasswordHasher passwordHasher;
+        switch (algorithm.toLowerCase()) {
+            case BCRYPT:
+                passwordHasher = getBCryptHasher(settings);
+                break;
+            case PBKDF2:
+                passwordHasher = getPBKDF2Hasher(settings);
+                break;
+            default:
+                throw new IllegalArgumentException("Password hashing algorithm not supported");
```

Comment:
can we log the "unsupported" algorithm being requested here? it will help to debug issues faster.

Reply:
Sorry, missed this comment initially! Will make the change.